### ISSUE TITLE
ZK 8.5.1  > TypeError: a.getTimeZone is not a function

### ISSUE THREAD

**Issue by adriens** [zkoss/zkcalendar#3]
Hi, Since we have upgrade to latest ZK 8.5.1 from 8.5.0, ZK Calendar does not work anymore, complaining about "TypeError: a.getTimeZone is not a function"

![outils de developpement - localhost8181 - httplocalhost8181](https://user-images.githubusercontent.com/5235127/37382189-83912b14-2795-11e8-82b0-a51f1720dcf8.jpg)

Details

It's totally linked to ZK 8.5.0->8.5.1 as when we go back to 8.5.0 things are running fine.

We arre currently running on the freshly version dependance :

<zk.calendar.version>2.1.6.FL.20161111</zk.calendar.version>

I've seen that Calendar is stuck on that version (https://github.com/zkoss/zkcalendar) with 0 activity since 2016..which seems a been scary to us : is it still maintained ?...or are we really using the good dependency ?

<dependency>
<groupId>org.zkoss.calendar</groupId>
<artifactId>calendar</artifactId>
<version>${zk.calendar.version}</version>
</dependency>

Consequence :

We cannot upgrade to ZK 8.5.1 : we are stuck on 8.5.0.

Thank you in advance for your help.

Best Regards,

Adrien

**Comment by cor3000** [comment]
Not a direct issue with ZKCalendar, side effect of [ZK-3527](http://tracker.zkoss.org/browse/ZK-3527)
To be fixed in 8.5.1.1 with [ZK-3897](http://tracker.zkoss.org/browse/ZK-3897)

Current workaround:
```
<script><![CDATA[
/*Temporary Patch for ZK-3897*/
zk.afterLoad('zul.db', function() {
   var xDate = {};
   zk.override(zk.fmt.Date, xDate, {
      formatDate : function(val, fmt, localizedSymbols) {
         if(val instanceof Date) {
            val.getTimeZone = zk.$void;
         }
         return xDate.formatDate.call(this, val, fmt, localizedSymbols);
      }
   });//zk.override
});//zk.afterLoad
]]></script>

```

**Comment by adriens** [comment]
WOW, thanks a lot for this good new @cor3000 !!! We'll wait for the 8.5.1.1 and **hence jump straight** from  `8.5.0` -> `8.5.1.1`


That issue was really managed like a :rocket:  :+1:
